### PR TITLE
voltron consumes from a topic per dataset

### PR DESCRIPTION
**Pull request by paul-jacob**

### CODE REVIEW

File: lib/voltron.ex [hunk 1]
```diff
@@ -0,0 +1,13 @@
+defmodule Voltron do
+  @moduledoc """
+  Main business logic of Forklift. Writes batches of messages to PrestoDB.
+  """
+  alias SmartCity.Data
+
+  @spec handle_batch(list(%Data{})) :: :ok | no_return()
```
Comment (cameronmarsh) [inline lib/voltron.ex]: This spec needs to be updated to SmartCirty.Data struct - Paul

Comment (paul-jacob) [reply lib/voltron.ex]: Done

File: lib/voltron/topic_manager.ex [hunk 2]
```diff
@@ -1,92 +1,45 @@
...
-  @kafka_timeout Application.get_env(:forklift, :kafka_timeout, 5_000)
-
-  defmodule Error do
-    defexception [:code, :message]
+  use Retry
+
+  def create_and_subscribe(topic, opts \\ []) do
+    Elsa.create_topic(endpoints(), topic, opts)
+
+    retry with: [10] |> Stream.cycle() |> Stream.take(10) do
```
Comment (ManApart) [inline lib/voltron/topic_manager.ex]: Why are we retrying with a cycle of 10? Is this a way to do an exponential backoff somehow?

File: lib/voltron/util.ex [hunk 3]
```diff
@@ -0,0 +1,48 @@
+defmodule Voltron.Util do
+  @moduledoc """
+  Utilities used inside voltron
+  """
+
+  def add_to_metadata(datum, field, value) do
```
Comment (ManApart) [inline lib/voltron/util.ex]: Seems like this would be a good candidate to move to Smartcity.Data

**Comment by jdenen** [comment]
What is `Smrt 1350`?

File: lib/voltron.ex [hunk 4]
```diff
@@ -0,0 +1,13 @@
+defmodule Voltron do
+  @moduledoc """
+  Main business logic of Forklift. Writes batches of messages to PrestoDB.
```
Comment (jakeprem) [inline lib/voltron.ex]: This moduledoc is still addressing Forklift

File: test/integration/kafka_integration_test.exs [hunk 5]
```diff
@@ -8,10 +8,12 @@ defmodule Voltron.KafkaIntegrationTest do
   alias SmartCity.TestDataGenerator, as: TDG
 
   import Record, only: [defrecord: 2, extract: 2]
-  defrecord :kafka_message, extract(:kafka_message, from_lib: "kafka_protocol/include/kpro_public.hrl")
 
-  @endpoint Application.get_env(:kaffe, :consumer)[:endpoints]
-            |> Enum.map(fn {k, v} -> {k, v} end)
+  defrecord :kafka_message,
```
Comment (jakeprem) [inline test/integration/kafka_integration_test.exs]: How are we using this record? Can this logic be pushed into Elsa somewhere? I don't think our apps should know about the Kafka Protocol library, especially not at the level of detail that includes a file path to a `.hrl` file.

File: lib/voltron/topic_manager.ex (hunk 2 above)
Comment (LtChae) [reply lib/voltron/topic_manager.ex]: This tries 10 times at 10ms each. There's no backoff.

File: test/integration/kafka_integration_test.exs (hunk 5 above)
Comment (LtChae) [reply test/integration/kafka_integration_test.exs]: It's a test, is that really a concern?

File: lib/voltron/util.ex (hunk 3 above)
Comment (LtChae) [reply lib/voltron/util.ex]: Agreed.

File: lib/voltron.ex (hunk 4 above)
Comment (LtChae) [inline lib/voltron.ex]:
```suggestion
  Main business logic of Voltron. Transforms messages and writes them to the next topic.
```

**Pull request merged**